File: PMUG02_16_2021_Project/Start_Java/app/src/main/java/com/neudesic/pmugfebappstart/PersonAdapter.java
package com.neudesic.pmugfebappstart;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.neudesic.pmugfebappstart.model.Person;

import java.util.List;

public class PersonAdapter extends RecyclerView.Adapter<PersonAdapter.PersonViewHolder> {

    private List<Person> people;

    public PersonAdapter(List<Person> people) {
        this.people = people;
    }

    @NonNull
    @Override
    public PersonViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new PersonViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_view_person, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull PersonViewHolder holder, int position) {
        Person person = people.get(position);

        holder.name.setText(person.getName());
        holder.age.setText(Integer.toString(person.getAge()));
        holder.birthDate.setText(person.getBirthday().toString());

    }

    @Override
    public int getItemCount() {
        return people.size();
    }

    public static class PersonViewHolder extends RecyclerView.ViewHolder {
        TextView name;
        TextView age;
        TextView birthDate;

        public PersonViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
            age = itemView.findViewById(R.id.age);
            birthDate = itemView.findViewById(R.id.birthdate);
        }
    }

    public void update(List<Person> people) {
        this.people = people;
    }
}

File: PMUG02_16_2021_Project/Start_Kotlin/app/src/main/java/com/neudesic/pmug02_16_2021_kotlin/PersonAdapter.kt
package com.neudesic.pmug02_16_2021_kotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.neudesic.pmug02_16_2021_kotlin.model.Person

class PersonAdapter(private var people: List<Person>): RecyclerView.Adapter<PersonAdapter.PersonViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PersonViewHolder {
        return PersonViewHolder(
                LayoutInflater.from(parent.context).inflate(R.layout.item_view_person, parent, false)
        )
    }

    override fun onBindViewHolder(holder: PersonViewHolder, position: Int) {
        val person: Person = people.get(position)
        holder.name.text = person.name
        holder.age.text = person.age.toString()
        holder.birthDate.text = person.birthday.toString()
    }

    override fun getItemCount(): Int {
        return people.size
    }

    class PersonViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var name: TextView = itemView.findViewById(R.id.name)
        var age: TextView = itemView.findViewById(R.id.age)
        var birthDate: TextView = itemView.findViewById(R.id.birthdate)

    }

    fun update(people: List<Person>) {
        this.people = people
    }
}